static/js/browser2/observation_chart.tsx

```diff
@@ -229,14 +230,14 @@ export class ObservationChart extends React.Component<
 
   private loadSpinner(): void {
     document
-      .getElementById(this.svgContainerId() + "-content-area")
+      .getElementById(this.containerId + "-content-area")
```

Comment:
maybe use a helper or constant for building that id to reduce future errors?

Reply:
I use the shared util function randDomId() to create the id

Reply:
yes, but you're also doing the same string concat multiple times. a shared util that will do that string concat, or saving it in a class var, will help.